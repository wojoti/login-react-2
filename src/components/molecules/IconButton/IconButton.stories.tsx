import Icon, { IconType } from "@components/atoms/Icon/Icon";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import IconButton from "./IconButton";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Molecules/IconButton",
  component: IconButton,
} as ComponentMeta<typeof IconButton>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof IconButton> = (args) => (
  <IconButton {...args} />
);

export const Google = Template.bind({});
Google.args = {
  color: "#DE5246",
  children: <Icon src={IconType.google} width={13} height={13} alt="google" />,
};

export const Facebook = Template.bind({});
Facebook.args = {
  color: "#1877f2",
  children: (
    <Icon src={IconType.facebook} width={13} height={13} alt="facebook" />
  ),
};

export const LinkedIn = Template.bind({});
LinkedIn.args = {
  color: "#0077B5",
  children: (
    <Icon src={IconType.linkedin} width={13} height={13} alt="linkedin" />
  ),
};
