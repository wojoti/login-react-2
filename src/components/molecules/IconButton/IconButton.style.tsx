import styled from "styled-components";

interface Props {
  accent?: string;
}
const Wrapper = styled.button<Props>`
  box-sizing: border-box;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 2px solid black;
  border-radius: 45px;
  padding: 3px 5px;
  border-color: ${(props) => props.accent || "black"};
  background-color: white;
  margin: 0px 10px;
  cursor: pointer;
`;
export default Wrapper;
