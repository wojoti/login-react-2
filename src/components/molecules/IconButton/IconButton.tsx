import { forwardRef, ReactNode } from "react";
import Wrapper from "./IconButton.style";

export interface IconButtonProps {
  children: ReactNode;
  color?: string;
  onClick: () => void;
  testId?: string;
}
export type IconButtonRef = HTMLButtonElement;

const IconButton = forwardRef<IconButtonRef, IconButtonProps>(
  ({ children, color, onClick, testId }, ref) => (
    <Wrapper accent={color} onClick={onClick} data-testid={testId} ref={ref}>
      {children}
    </Wrapper>
  )
);
IconButton.displayName = "IconButton";

export default IconButton;
