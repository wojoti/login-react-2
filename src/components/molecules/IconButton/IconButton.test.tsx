import { fireEvent, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import IconButton, { IconButtonProps } from "./IconButton";

const handleClick = jest.fn();
const props: IconButtonProps = {
  testId: "test-iconbutton-id",
  children: "test iconbutton",
  onClick: handleClick,
};

const modifiedProps: IconButtonProps = {
  ...props,
  color: "blue",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<IconButton {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render iconbutton", () => {
  render(<IconButton {...props} />);
  const iconbuttonElement = screen.getByTestId("test-iconbutton-id");
  expect(iconbuttonElement).toBeInTheDocument();
});

test("should handle onClick event", () => {
  render(<IconButton {...props} />);
  const iconbuttonElement = screen.getByTestId("test-iconbutton-id");
  fireEvent.click(iconbuttonElement);
  expect(handleClick).toHaveBeenCalledTimes(1);
});

test("should render iconbutton with modified style through props", () => {
  render(<IconButton {...modifiedProps} />);
  const iconbuttonElement = screen.getByTestId("test-iconbutton-id");
  expect(iconbuttonElement).toBeInTheDocument();
  expect(iconbuttonElement).toHaveStyle("border-color: blue");
});
