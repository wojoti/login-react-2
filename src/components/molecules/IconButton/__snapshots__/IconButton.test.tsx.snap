// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 2px solid black;
  border-radius: 45px;
  padding: 3px 5px;
  border-color: black;
  background-color: white;
  margin: 0px 10px;
  cursor: pointer;
}

<button
  className="c0"
  data-testid="test-iconbutton-id"
  onClick={[MockFunction]}
>
  test iconbutton
</button>
`;
